android/src/main/java/org/mozilla/osmdroid/tileprovider/modules/SmartFSTileLoader.java

```diff
@@ -0,0 +1,84 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.osmdroid.tileprovider.modules;
+
+import android.graphics.drawable.Drawable;
+
+import org.mozilla.mozstumbler.service.AppGlobals;
+import org.mozilla.mozstumbler.service.core.logging.Log;
+import org.mozilla.osmdroid.tileprovider.MapTile;
+import org.mozilla.osmdroid.tileprovider.MapTileRequestState;
+import org.mozilla.osmdroid.tileprovider.constants.OSMConstants;
+import org.mozilla.osmdroid.tileprovider.tilesource.BitmapTileSourceBase;
+import org.mozilla.osmdroid.tileprovider.tilesource.ITileSource;
+
+import java.io.File;
+
+class SmartFSTileLoader extends AbstractTileLoader {
+
+    private final static String LOG_TAG = AppGlobals.makeLogTag(SmartFSTileLoader.class);
+    private SmartFSProvider smartFSProvider;
+
+    public SmartFSTileLoader(SmartFSProvider mapTileModuleProviderBase) {
+        super(mapTileModuleProviderBase);
+        smartFSProvider = mapTileModuleProviderBase;
+    }
+
+
+    @Override
+    public Drawable loadTile(final MapTileRequestState pState) throws CantContinueException {
+
+        ITileSource tileSource = smartFSProvider.getTileSource();
+
+        if (tileSource == null) {
+            return null;
+        }
+
+        final MapTile tile = pState.getMapTile();
+
+        // if there's no sdcard then don't do anything
+        if (!smartFSProvider.getSdCardAvailable()) {
```

Comment:
That's the same code that was in the TileLoader before.  I just moved the code and didn't want to add any functional changes in this patch.